financial-connections/src/main/java/com/stripe/android/financialconnections/presentation/FinancialConnectionsViewModel.kt

```diff
@@ -0,0 +1,18 @@
+package com.stripe.android.financialconnections.presentation
+
+import com.airbnb.mvrx.MavericksState
+import com.airbnb.mvrx.MavericksViewModel
+import com.stripe.android.financialconnections.navigation.topappbar.TopAppBarHost
+
+@Suppress("UNUSED_PARAMETER", "UnnecessaryAbstractClass")
+internal abstract class FinancialConnectionsViewModel<S : MavericksState>(
+    initialState: S,
+    topAppBarHost: TopAppBarHost,
+) : MavericksViewModel<S>(initialState) {
+
+    init {
+        onEach { state ->
```

Comment:
Given we'll be dealing with a generic State class here, how would this logic look like? Would the idea to have a base state class with some common property(s)?

Reply:
It’ll look like this in the follow-up pull request https://github.com/stripe/stripe-android/pull/8135: 

https://github.com/stripe/stripe-android/blob/dbd3d586f700b38f5732c5767dc7fc51c0f6c112/financial-connections/src/main/java/com/stripe/android/financialconnections/presentation/FinancialConnectionsViewModel.kt